Add Docker setup guide - Closes #1163 (#1227)
LiskHQ/lisk-docs: modules/ROOT/pages/setup/docker.adoc

= Docker setup
Mona Bärenfänger <mona@lightcurve.io>
:description: How to install and setup up a Lisk Core node with Docker.
:toc:
// Settings
// External URLs
:url_core_releases: https://github.com/LiskHQ/lisk-core/releases
:url_podman: https://github.com/containers/podman/
:url_docker_install: https://docs.docker.com/engine/installation/#desktop
:url_docker_install_linux: https://docs.docker.com/engine/installation/#server
:url_docker_install_mac: https://docs.docker.com/docker-for-mac/install/
:url_docker_install_windows: https://docs.docker.com/docker-for-windows/install/
:url_docker_linux_post_install: https://docs.docker.com/install/linux/linux-postinstall
:url_xcode: https://developer.apple.com/xcode/features/
// Project URLs
:url_config: management/configuration.adoc
:url_upgrade_source: update/source.adoc

> This document covers how to prepare a system to run Lisk Core as a Docker image-based container.

Reviewer: ```suggestion
This document describes how to build a system to run Lisk Core as a Docker image-based container.
```
or
```suggestion
This document describes how to run Lisk Core as a Docker image-based container.
```

Lisk Core version 3 does not have any external dependencies and thus does not require using `docker-compose`.

== Pre-installation

=== Create a new user `lisk`

[source,bash]
----
adduser lisk
----

Add the `lisk` user to the `sudo` user group:

[source,bash]
----
sudo usermod -aG sudo lisk
----

=== Install Docker

> To run Lisk in Docker a user must first install the Docker engine.

Reviewer: ```suggestion
To run Lisk Core in Docker, a user must first install the Docker engine.
```

Author: ```suggestion
To run Lisk Core in Docker, it is necessary to first install the Docker engine.
```
